opendut-edgar/src/setup/tasks/write_configuration.rs

```diff
@@ -136,16 +168,69 @@ mod tests {
     use predicates::prelude::predicate;
     use uuid::uuid;
     use googletest::prelude::*;
+    use opendut_types::util::net::{ClientId, ClientSecret, OAuthScope};
 
     use crate::setup::runner;
 
     use super::*;
 
     const HOST: &str = "example.com";
     const PORT: u16 = 1234;
+    const CLIENT_ID: &str = "ClientId";
+    const CLIENT_SECRET: &str = "ClientSecret";
+    const OIDC_ENABLED: bool = true;
+
+    #[test]
+    fn should_write_a_fresh_configuration_with_auth_config_enabled() -> anyhow::Result<()> {
+        let temp = TempDir::new()?;
+        let config_file = temp.child("edgar.toml");
+        let config_merge_suggestion_file = temp.child("edgar-merge-suggestion.toml");
+
+        let new_peer_id = PeerId::from(uuid!("dc72f6d9-d700-455f-8c31-9f15438e7503"));
+
+        let task = WriteConfiguration {
```

Comment:
There are multiple duplicate code warnings for this file. What about creating a test fixture for those shared parts?

Reply:
Use the `#[fixture]` macro from rstest_macros to create a struct for structs that are used multiple times.

Reply:
Great. This reduces repetition and now you have used fixtures in rust as well! :-)